packages/machine/test/integration/install-then-set-state.spec.ts

```diff
@@ -0,0 +1,239 @@
+import AppRegistry from "@counterfactual/contracts/build/contracts/AppRegistry.json";
+import ETHBucket from "@counterfactual/contracts/build/contracts/ETHBucket.json";
+import MinimumViableMultisig from "@counterfactual/contracts/build/contracts/MinimumViableMultisig.json";
+import MultiSend from "@counterfactual/contracts/build/contracts/MultiSend.json";
+import NonceRegistry from "@counterfactual/contracts/build/contracts/NonceRegistry.json";
+import ProxyFactory from "@counterfactual/contracts/build/contracts/ProxyFactory.json";
+import StateChannelTransaction from "@counterfactual/contracts/build/contracts/StateChannelTransaction.json";
+import { AssetType, NetworkContext } from "@counterfactual/types";
+import dotenv from "dotenv-safe";
+import { Contract, Wallet } from "ethers";
+import { AddressZero, WeiPerEther, Zero } from "ethers/constants";
+import { JsonRpcProvider } from "ethers/providers";
+import {
+  BigNumber,
+  hexlify,
+  Interface,
+  parseEther,
+  randomBytes,
+  SigningKey
+} from "ethers/utils";
+
+import {
+  InstallCommitment,
+  SetStateCommitment
+} from "../../src/middleware/protocol-operation";
+import { AppInstance, StateChannel } from "../../src/models";
+
+// To be honest, 30000 is an arbitrary large number that has never failed
+// to reach the done() call in the test case, not intelligency chosen
+const JEST_TEST_WAIT_TIME = 30000;
+
+// ProxyFactory.createProxy uses assembly `call` so we can't estimate
+// gas needed, so we hard-code this number to ensure the tx completes
+const CREATE_PROXY_AND_SETUP_GAS = 6e9;
+
+// The AppRegistry.setState call _could_ be estimated but we haven't
+// written this test to do that yet
+const SETSTATE_COMMITMENT_GAS = 6e9;
+
+// Also we can't estimate the install commitment gas b/c it uses
+// delegatecall for the conditional transaction
+const INSTALL_COMMITMENT_GAS = 6e9;
+
+let networkId: number;
+let provider: JsonRpcProvider;
+let wallet: Wallet;
+let network: NetworkContext;
+
+// TODO: When we add a second use case of this custom mathcer,
+//       move it and its typigns into somewhere re-usable
+declare global {
+  namespace jest {
+    interface Matchers<R> {
+      toBeEq(expected: BigNumber): BigNumber;
+    }
+  }
+}
+expect.extend({
+  toBeEq(received: BigNumber, argument: BigNumber) {
+    return {
+      pass: received.eq(argument),
+      message: () => `expected ${received} not to be equal to ${argument}`
+    };
+  }
+});
+
+// TODO: This will be re-used for all integration tests, so
+//       move it somewhere re-usable when we add a new test
+beforeAll(async () => {
+  dotenv.config();
+
+  // Can use ! because dotenv-safe ensures value is set
+  const host = process.env.DEV_GANACHE_HOST!;
+  const port = process.env.DEV_GANACHE_PORT!;
+  const mnemonic = process.env.DEV_GANACHE_MNEMONIC!;
+
+  function pathForAccount(idx: number) {
+    return `m/44'/60'/0'/0/${idx}`;
+  }
```

Comment:
This uses the HD Path format to get the equivalent of `accounts[idx]` in Truffle tests. I did this when trying to get parallelism to work (avoiding transaction nonce being out of sync) but ended up not needing to do it.